appium/options/android/common/avd/avd_options.py

```diff
@@ -0,0 +1,210 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from appium.options.base_options import OptionsDescriptor
+from appium.options.common.supports_capabilities import SupportsCapabilities
+
+
+class AvdArgsOption(SupportsCapabilities):
+    AVD_ARGS = 'avdArgs'
+    avd_args = OptionsDescriptor('AVD_ARGS')
+    """
+    Gets and Sets emulator command line arguments.
+
+    Usage
+    -----
+    - Get
+        - `self.avd_args`
+    - Set
+        - `self.avd_args` = `value`
+    
+    Parameters
+    ----------
+    `value`: `str`
+
+    Returns
+    -------
+    - Get
+        - `Optional[str]`
+    - Set
+        - `None`
+    """
+
+
+class AvdEnvOption(SupportsCapabilities):
+    AVD_ENV = 'avdEnv'
+    avd_env = OptionsDescriptor('AVD_ENV')
+    """
+    Gets and Sets the mapping of emulator environment variables.
+
+    Usage
+    -----
+    - Get
+        - `self.avd_env`
+    - Set
+        - `self.avd_env` = `value`
+
+    Parameters
+    ----------
+    `value`: `str`
+
+    Returns
+    -------
+    - Get
+        - `Optional[Dict[str, str]`
+    - Set
+        - `None`
+    """
+
+
+class AvdLaunchTimeoutOption(SupportsCapabilities):
+    AVD_LAUNCH_TIMEOUT = 'avdLaunchTimeout'
+    avd_launch_timeout = OptionsDescriptor('AVD_LAUNCH_TIMEOUT')
+    """
+    Timeout to wait until Android Emulator is started.
+    Maximum timeout to wait until Android Emulator is started.
+    60000 ms by default.
+
+    Usage
+    -----
+    - Get
+        - `self.avd_launch_timeout`
+    - Set
+        - `self.avd_launch_timeout` = `value`
+    
+    Parameters
+    ----------
+    `value`: `Union[timedelta, int]`
+
+    Returns
+    -------
+    - Get
+        - `Optional[timedelta]`
+    - Set
+        - `None`
+    """
+
+
+class AvdOption(SupportsCapabilities):
+    AVD = 'avd'
+    avd = OptionsDescriptor('AVD')
+    """
+    The name of Android emulator to run the test on.
+    Names of currently installed emulators could be listed using
+    avdmanager list avd command. If the emulator with the given name
+    is not running then it is going to be started before a test.
+
+    Usage
+    -----
+    - Get
+        - `self.avd`
+    - Set
+        - `self.avd` = `value`
+    
+    Parameters
+    ----------
+    `value`: `str`
+
+    Returns
+    -------
+    - Get
+        - `Optional[str]`
+    - Set
+        - `None`
+    """
+
+class AvdReadyTimeoutOption(SupportsCapabilities):
+    AVD_READY_TIMEOUT = 'avdReadyTimeout'
+    avd_ready_timeout = OptionsDescriptor('AVD_READY_TIMEOUT')
```

Comment:
Transformers must be applied to all options where they are needed

Reply:
> Transformers must be applied to all options where they are needed

Yes. I have missed it. Thanks for letting me know. I will make all the changes, I will review it once again with the current release version and re-commit the code.